src/test/java/plugins/LogParserTest.java

```diff
@@ -63,6 +62,30 @@ public void trendVisible(){
     }
 
     /**
+     * Test case:
+     * Check for a build to be marked as failed and another build to be marked as unstable.
+     */
+    @Test
+    public void checkMarkedFailedAndUnstable() {
+        FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class, "simple-job");
+        job.configure(() -> {
+            job.addShellStep("echo marked as error");
+            LogParserPublisher lpp = job.addPublisher(LogParserPublisher.class);
+            lpp.setMarkOnBuildFail(true);
+            lpp.setRule(resource("/logparser_plugin/rules/log-parser-rule-markings"));
+        });
+        job.startBuild().waitUntilFinished().shouldFail();
+
+        job.configure(() -> {
+            job.addShellStep("echo marked as warning");
+            LogParserPublisher lpp = job.getPublisher(LogParserPublisher.class);
+            lpp.setMarkOnBuildFail(false);
+            lpp.setMarkOnUnstableWarning(true);
+        });
+        job.startBuild().waitUntilFinished().shouldBeUnstable();
+    }
+
+    /**
```

Comment:
What I actually meant was (to get rid of the duplication):
```
@Test
public void checkMarkedFailed() {
       FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class);
       configureBuildStatus(true, false);
       job.startBuild().waitUntilFinished().shouldFail();
}
@Test
public void checkMarkedUnstable() {
       FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class);
       configureBuildStatus(false, true);
       job.startBuild().waitUntilFinished().shouldBeUnstable();
}
```

Or even simpler:
@Test
public void checkMarkedFailed() {
       modifyBuildStatus(true, false, Result.FAILED);
}
@Test
public void checkMarkedUnstable() {
       modifyBuildStatus(false, true, Result.UNSTABLE);
}